### PR TITLE
Add local override options for GMT threshold when db is not populated

### PR DESCRIPTION
**Pull request by amponce**
## Summary
This PR introduces new options for manually overriding the Copay and  GMT (Geographical Mean Test) threshold API request with mock or staging data. It adds parameters that enable us to set mock data to true when running locally, facilitating a streamlined experience for quick testing. This update ensures a safe development process by preventing accidental commits that might enable mock data in production.

```js
// when USE_MOCK is true, the application will use the copay mock data below instead of live api data
export const USE_COPAY_MOCK_DATA = false;
// when USE_GEOGRAPHIC_MOCK_DATA is true, the application will use the geographic mock data below instead of live api data
export const USE_GEOGRAPHIC_MOCK_DATA = false;
// this is the value that will be used for GMT calculations when USE_GEOGRAPHIC_MOCK_DATA is true
export const MOCK_GMT_VALUE = 78300;
// when USE_STAGING_DATA is true, the application will use the staging api instead of the production api or USE_GEOGRAPHIC_MOCK_DATA
export const USE_STAGING_DATA = false;

```

## Related issue(s)
- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#70716

- _Link to related ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#69390

Testing done
- Prior to this change, developers had to rely solely on the production or staging APIs for testing, which could be time-consuming.
- To verify these changes, run the application locally with various settings of USE_MOCK_DATA and USE_STAGING_DATA.
- Tests were conducted to ensure that when USE_MOCK_DATA or USE_STAGING_DATA is set to true in a local environment, the application behaves as expected without affecting the staging or production environments.

### CODE REVIEW

**Review by powellkerry — APPROVED** [review]

**Comment by powellkerry** [comment]
Looks like there are some flaky tests associated with the app you are updating. They will need to be fixed before you are able to merge.

**Comment by amponce** [comment]
Thanks @powellkerry

Yeah I don't know what's going on today, I'm having trouble some identifying a fix for the flaky tests since they seem related to another team. 

Side note: I added a unit test redux state management after you made your comment, just wanted to give you a heads up

**Comment by amponce** [comment]
I'm out 11/14 for my daughters birthday so I'm taking this out of review mode until I return on 11/15

File: src/applications/financial-status-report/actions/geographicMeansThreshold.js [hunk 1]
```diff
@@ -1,48 +1,107 @@
...
+
+/**
+ * Determines the appropriate data source based on configuration settings and environment.
+ *
+ * @returns {string} The URL of the data source.
+ */
+const getDataUrl = (dependents, year, zipCode) => {
+  const isLocalhost = environment.isLocalhost();
+
+  if (isLocalhost && USE_GEOGRAPHIC_MOCK_DATA) {
```
[the remark this answers is not in the document]
Comment (amponce) [reply src/applications/financial-status-report/actions/geographicMeansThreshold.js:53]: USE_GEOGRAPHIC_MOCK_DATA is a manual toggle for development purposes.
It is intended to be set to 'true' by developers who need to use mock data.
This allows for easier local development when the necessary databases are not available.

File: src/applications/financial-status-report/actions/copays.js [hunk 2]
```diff
@@ -33,24 +42,43 @@
   });
 };
 
+// Mock function for retrieving statements data
+const getStatementsMock = async dispatch => {
+  if (!USE_COPAY_MOCK_DATA) return;
```
[the remark this answers is not in the document]
Comment (amponce) [reply src/applications/financial-status-report/actions/copays.js:47]: USE_STAGING_DATA is a manual toggle for development purposes.
It is intended to be set to 'true' by developers who need to use mock data.
This allows for easier local development when the necessary databases are not available.

File: src/applications/financial-status-report/actions/geographicMeansThreshold.js [hunk 3]
```diff
@@ -1,48 +1,107 @@
...
+    return null; // Return mock data directly
+  }
 
-const INCOME_UPPER_PERCENTAGE = 1.5;
-const ASSET_PERCENTAGE = 0.065;
-const DISCRETIONARY_INCOME_PERCENTAGE = 0.0125;
+  if (isLocalhost && USE_STAGING_DATA) {
```
[the remark this answers is not in the document]
Comment (amponce) [reply src/applications/financial-status-report/actions/geographicMeansThreshold.js:57]: USE_STAGING_DATA is a manual toggle for development purposes.
It is intended to be set to 'true' by developers who need to use mock data.
This allows for easier local development when the necessary databases are not available.

**Review by powellkerry — APPROVED** [review]

**Review by OptionSelect — APPROVED** [review]
LGTM! 🚀

File: src/applications/financial-status-report/actions/geographicMeansThreshold.js [hunk 4]
```diff
@@ -1,48 +1,107 @@
...
+ *
+ * @param {number} dependents - The number of dependents.
+ * @param {number} year - The year for which data is being requested.
+ * @param {string} zipCode - The ZIP code for which data is being requested.
+ * @returns {Promise<Object>} A promise resolving to the GMT thresholds data or error object.
+ */
+export const getGMT = async (dependents, year, zipCode) => {
+  const dataUrl = getDataUrl(dependents, year, zipCode);
+
+  if (dataUrl === null) {
```
Comment (Dr-Pongo) [inline src/applications/financial-status-report/actions/geographicMeansThreshold.js:78]: I know we talked about this already, but the `null` just gives me the heebie jeebies. Still going to slap an approval on it, but a soft suggestion of maybe setting `dataUrl` to a string like 'useMock' and have the check there

**Review by Dr-Pongo — APPROVED** [review]
Left a small note, but totally up to you!
LGTM! 🚀

File: src/applications/financial-status-report/actions/geographicMeansThreshold.js (hunk 4 above)
Comment (amponce) [reply src/applications/financial-status-report/actions/geographicMeansThreshold.js]: Added  a check of isLocalhost for an extra layer of protection 👍

**Pull request merged**